### PR TITLE
Give the online lua mainmenu also the client_list and mods

### PR DESCRIPTION
**Pull request by Lejo1**
The lua Mainmenu interface now gets also the clients and the mods!
This allows new Features just like listing players(maybe tooltip) or searching.
Finding servers with the wanted mods and more...

### CODE REVIEW

File: src/script/lua_api/l_mainmenu.cpp [hunk 1]
```diff
@@ -391,6 +391,37 @@ int ModApiMainMenu::l_get_favorites(lua_State *L)
 			lua_settable(L, top_lvl2);
 		}
 
+		if (server["clients_list"].isArray()) {
+			unsigned int index_lvl2 = 1;
+			lua_pushstring(L,"clients_list");
```
Comment (SmallJoker) [inline src/script/lua_api/l_mainmenu.cpp]: Space after comma. The old code wasn't checked for code style.
https://dev.minetest.net/Code_style_guidelines

**Comment by SmallJoker** [comment]
Sure, why not? Please update the docs accordingly:
https://github.com/minetest/minetest/blob/a6f784835244111c90c05b6aae47092c2c5763a7/doc/menu_lua_api.txt#L153-L155

**Comment by Lejo1** [comment]
Done.

**Comment by Enrikoo** [comment]
I have no Idea how this works I have tried to search for clients and It doesnt find. So how does it work?

**Comment by Lejo1** [comment]
> I have no Idea how this works I have tried to search for clients and It doesnt find. So how does it work?

It's just a way this change can be used for.

**Comment by Enrikoo** [comment]
So I tried to find for players with the search button but it didnt list players, only servers. How can I find for something different?

**Comment by Lejo1** [comment]
> So I tried to find for players with the search button but it didnt list players, only servers. How can I find for something different?

You understand me wrong. You can't find a player list. But maybe a playerlist will be added soon.

**Comment by Enrikoo** [comment]
Yes I cant understand this, so only the code has been changed or does it recognize game versions better, but it does not have to do with mainmenu.

**Comment by Lejo1** [comment]
It **maybe** if added will be available in the next release.

**Comment by Enrikoo** [comment]
Okay I have to wait, I can't do it yet.

**Pull request merged**